core/src/test/scala/org/apache/spark/sql/delta/SnapshotManagementSuite.scala

```diff
@@ -304,4 +305,39 @@ class SnapshotManagementSuite extends QueryTest with SQLTestUtils with SharedSpa
         expectedEndVersion = None)
     }
   }
+
+  test("configurable snapshot cache storage level") {
+    withTempDir { tempDir =>
+      val path = tempDir.getCanonicalPath
+      spark.range(10).write.format("delta").save(path)
+      DeltaLog.clearCache()
+      assert(sparkContext.getPersistentRDDs.isEmpty)
```

Comment:
Since unpersisting RDDs may not be as quick as possible, it's better to just search for the RDDs we are looking for by their names so that this test won't be flaky: https://github.com/delta-io/delta/blob/0ed90b5822cd23c3d44b1a685d90f66cc741362e/core/src/main/scala/org/apache/spark/sql/delta/Snapshot.scala#L152

Reply:
Are you referring to the delta side of the clear cache or the spark side of the unpersist? Because it looks like the RDD is removed from persistentRdds synchronously

Reply:
@Kimahriman you are right. I was thinking that removing blocks is asynchronous as we call `_.unpersist(blocking = false)`. But `getPersistentRDDs` will be updated synchronously.

Reply:
Yeah I had to look it up myself to make sure, I guess it's just the messages to the executors that are non-blocking